service/crawlingKeywordService.js

```diff
@@ -79,4 +87,40 @@ const getCrawlingKeyword = async (req, res) => {
   }
 };
 
+const isCheckTrueThisUrl = (url) => {
+  /* eslint-disable */
+  const urlRegex = /^(http|https):\/\/[\w\-]+(\.[\w\-]+)+[/#?]?.*$/;
+  if (urlRegex.test(url)) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const getAllSentence = (innerText) => {
+  return innerText
+    .replace(/\n|\r|\t/g, " ")
+    .split(/(?<=다\. |요\. |니다\. |\. |! |\? )/)
+    .map((sentence) => sentence.trim())
+    .filter((sentence) => sentence);
+};
```

Comment:
split으로 배열 분리 후 filter를 통해 빈 값을 제외하고 return을 해주는 코드 같습니다.
제 생각으론 이 코드의 개선점으로 반복을 두 번이나 돌 필요가 있을까 입니다.
그래서 저는 아래와 같이 1회의 반복으로 공백제거와 공백값을 제외를 동시에 하는 것이 더욱 효율적이라고 생각이 드는데
아래의 코드 참고해보시고 어떠신지 생각 남겨주시기 바랍니다.
```suggestion
const getAllSentence = (innerText) => {
  return innerText
    .replace(/\n|\r|\t/g, " ")
    .split(/(?<=다\. |요\. |니다\. |\. |! |\? )/)
    .map((sentence) => {
      const trimedSentence = sentence.trim();
      
      if (trimedSentence) {
        return sentence;
      }
    })
};
```

Reply:
저 또한 굳이 filter로 인해 두 번 반복되지 않고 map을 통한 한 번의 반복으로 원하는 데이터를 만들 수 있다면 성능 향상에도 도움이 될 것 같다고 생각합니다. 좋은 의견 감사합니다! 수정하여 반영하겠습니다.